programs/spring-util/actions/blocklog.cpp

```diff
@@ -323,27 +298,74 @@ int blocklog_actions::read_log() {
          *out << fc::json::to_pretty_string(v) << "\n";
    };
    bool contains_obj = false;
-   while((block_num <= opt->last_block) && (next = block_logger.read_block_by_num(block_num))) {
-      if(opt->as_json_array && contains_obj)
-         *out << ",";
-      print_block(next);
-      ++block_num;
-      contains_obj = true;
-   }
 
-   if(!fork_db_branch.empty()) {
-      for(auto bitr = fork_db_branch.rbegin(); bitr != fork_db_branch.rend() && block_num <= opt->last_block; ++bitr) {
-         if(opt->as_json_array && contains_obj)
+   bool print_log     = opt->print_from == print_from_t::both || opt->print_from == print_from_t::block_log;
+   bool print_fork_db = opt->print_from == print_from_t::both || opt->print_from == print_from_t::fork_db;
+
+   signed_block_ptr end;
+
+   report_time rt(print_log && print_fork_db ? "reading both log and fork_db"
+                                             : (print_log ? "reading log" : "reading fork_db"));
+   if (print_log) {
+      block_log   block_logger(opt->blocks_dir, opt->blog_conf);
+      end = block_logger.read_head();
+      EOS_ASSERT(end, block_log_exception, "No blocks found in block log");
+      EOS_ASSERT(end->block_num() > 1, block_log_exception, "Only one block found in block log");
+
+      ilog("existing block log contains block num ${first} through block num ${n}",
+           ("first", block_logger.first_block_num())("n", end->block_num()));
+      
+      block_num = std::max(block_num, block_logger.first_block_num());
+
+      while ((block_num <= opt->last_block) && (next = block_logger.read_block_by_num(block_num))) {
+         if (opt->as_json_array && contains_obj)
             *out << ",";
-         auto& next = *bitr;
          print_block(next);
          ++block_num;
          contains_obj = true;
       }
    }
 
+   if (print_fork_db) {
+      block_branch_t fork_db_branch;
+      if (std::filesystem::exists(std::filesystem::path(opt->blocks_dir) / config::reversible_blocks_dir_name /
+                                  config::fork_db_filename)) {
+         ilog("opening fork_db");
```

Comment:
Is this ilog necessary?

Reply:
It is not strictly necessary, but I think it doesn't hurt either. I can remove it if you think having it is objectionable.

Reply:
No, you don't need to change.